lib/medtronic600/NGPHistoryParser.js

```diff
@@ -0,0 +1,894 @@
+      EASY_BOLUS_OPTION_CHANGE: 0x66,
+      BG_REMINDER_OPTION_CHANGE: 0x68,
+      BG_REMINDER_TIME: 0x69,
+      AUDIO_VIBRATE_MODE_CHANGE: 0x6A,
+      TIME_FORMAT_CHANGE: 0x6B,
+      LOW_RESERVOIR_WARNING_CHANGE: 0x6C,
+      LANGUAGE_CHANGE: 0x6D,
+      STARTUP_WIZARD_START_END: 0x6E,
+      REMOTE_BOLUS_OPTION_CHANGE: 0x6F,
+      AUTO_SUSPEND_CHANGE: 0x72,
+      BOLUS_DELIVERY_RATE_CHANGE: 0x73,
+      DISPLAY_OPTION_CHANGE: 0x77,
+      SET_CHANGE_REMINDER_CHANGE: 0x78,
+      BLOCK_MODE_CHANGE: 0x79,
+      BOLUS_WIZARD_SETTINGS_SUMMARY: 0x7B,
+      SENSOR_SETTINGS_CHANGE: 0xC8,
+      OLD_SENSOR_WARNING_LEVELS: 0xC9,
+      NEW_SENSOR_WARNING_LEVELS: 0xCA,
+      GENERAL_SENSOR_SETTINGS_CHANGE: 0xCB,
+      SENSOR_GLUCOSE_READINGS: 0xCC,
+      SENSOR_GLUCOSE_GAP: 0xCD,
+      GLUCOSE_SENSOR_CHANGE: 0xCE,
+      SENSOR_CALIBRATION_REJECTED: 0xCF,
+      SENSOR_ALERT_SILENCE_STARTED: 0xD0,
+      SENSOR_ALERT_SILENCE_ENDED: 0xD1,
+      OLD_LOW_SENSOR_WARNING_LEVELS: 0xD2,
+      NEW_LOW_SENSOR_WARNING_LEVELS: 0xD3,
+      OLD_HIGH_SENSOR_WARNING_LEVELS: 0xD4,
+      NEW_HIGH_SENSOR_WARNING_LEVELS: 0xD5,
+      SENSOR_GLUCOSE_READINGS_EXTENDED: 0xD6,
+      NORMAL_BOLUS_DELIVERED: 0xDC,
+      SQUARE_BOLUS_DELIVERED: 0xDD,
+      DUAL_BOLUS_PART_DELIVERED: 0xDE,
+    };
+  }
+
+  constructor(eventData) {
+    this.eventData = eventData;
+  }
+
+  get source() {
+    // No idea what "source" means.
+    return this.eventData[0x01];
+  }
+
+  get size() {
+    return this.eventData[0x02];
+  }
+
+  get eventType() {
+    return this.eventData[0];
+  }
+
+  eventInstance() {
+    /* eslint-disable no-use-before-define */
+    switch (this.eventType) {
+      case NGPHistoryEvent.EVENT_TYPE.SENSOR_GLUCOSE_READINGS_EXTENDED:
+        return new SensorGlucoseReadingsEvent(this.eventData);
+      case NGPHistoryEvent.EVENT_TYPE.BG_READING:
+        return new BloodGlucoseReadingEvent(this.eventData);
+      case NGPHistoryEvent.EVENT_TYPE.BASAL_SEGMENT_START:
+        return new BasalSegmentStartEvent(this.eventData);
+      case NGPHistoryEvent.EVENT_TYPE.TEMP_BASAL_COMPLETE:
+        return new TempBasalCompleteEvent(this.eventData);
+      case NGPHistoryEvent.EVENT_TYPE.REWIND:
+        return new RewindEvent(this.eventData);
+      case NGPHistoryEvent.EVENT_TYPE.CANNULA_FILL_DELIVERED:
+        return new CannulaFillDeliveredEvent(this.eventData);
+      case NGPHistoryEvent.EVENT_TYPE.NORMAL_BOLUS_DELIVERED:
+        return new NormalBolusDeliveredEvent(this.eventData);
+      case NGPHistoryEvent.EVENT_TYPE.SQUARE_BOLUS_DELIVERED:
+        return new SquareBolusDeliveredEvent(this.eventData);
+      case NGPHistoryEvent.EVENT_TYPE.DUAL_BOLUS_PART_DELIVERED:
+        return new DualBolusPartDeliveredEvent(this.eventData);
+      case NGPHistoryEvent.EVENT_TYPE.BOLUS_WIZARD_ESTIMATE:
+        return new BolusWizardEstimateEvent(this.eventData);
+      default:
+        // debug('Unknown Instance type:', this.eventType);
```

Comment:
Can we already remove this commented-out debug statement, or do you plan on maybe using it in future?